packages/openapi-to-graphql/src/oas_3_tools.ts

```diff
@@ -125,55 +125,55 @@ export function methodToHttpMethod(method: string): HTTP_METHODS {
   }
 }
 
+export function isOas2(spec: any): spec is Oas2 {
+  return typeof spec.swagger === 'string' && /^2/.test(spec.swagger)
+}
+
+export function isOas3(spec: any): spec is Oas3 {
+  return typeof spec.openapi === 'string' && /^3/.test(spec.openapi)
+}
+
 /**
  * Resolves on a validated OAS 3 for the given spec (OAS 2 or OAS 3), or rejects
  * if errors occur.
  */
-export function getValidOAS3(
+export async function getValidOAS3(
   spec: Oas2 | Oas3,
   oasValidatorOptions: object,
   swagger2OpenAPIOptions: object
 ): Promise<Oas3> {
-  return new Promise((resolve, reject) => {
-    // CASE: translate
-    if (
-      typeof (spec as Oas2).swagger === 'string' &&
-      (spec as Oas2).swagger === '2.0'
-    ) {
-      preprocessingLog(
-        `Received Swagger - going to translate to OpenAPI Specification...`
+  // CASE: translate
+  if (isOas2(spec)) {
+    preprocessingLog(
+      `Received Swagger - going to translate to OpenAPI Specification...`
+    )
+    try {
+      const { openapi } = await Swagger2OpenAPI.convertObj(
+        spec,
+        swagger2OpenAPIOptions
+      )
+      return openapi
+    } catch (error) {
+      throw new Error(
+        `Could not convert Swagger '${spec.info.title}' to OpenAPI Specification. ${error.message}`
       )
-
-      Swagger2OpenAPI.convertObj(spec, swagger2OpenAPIOptions)
-        .then((options) => resolve(options.openapi))
-        .catch((error) =>
-          reject(
-            `Could not convert Swagger '${
-              (spec as Oas2).info.title
-            }' to OpenAPI Specification. ${error.message}`
-          )
-        )
-
-      // CASE: validate
-    } else if (
-      typeof (spec as Oas3).openapi === 'string' &&
-      /^3/.test((spec as Oas3).openapi)
-    ) {
-      preprocessingLog(`Received OpenAPI Specification - going to validate...`)
-
-      OASValidator.validate(spec, oasValidatorOptions)
-        .then(() => resolve(spec as Oas3))
-        .catch((error) =>
-          reject(
-            `Could not validate OpenAPI Specification '${
-              (spec as Oas3).info.title
-            }'. ${error.message}`
-          )
-        )
-    } else {
-      reject(`Invalid specification provided`)
     }
-  })
+    // CASE: validate
+  } else if (isOas3(spec)) {
+    preprocessingLog(`Received OpenAPI Specification - going to validate...`)
+    try {
+      await OASValidator.validate(spec, oasValidatorOptions)
+    } catch (error) {
+      console.warn(
+        `Could not validate OpenAPI Specification '${
+          (spec as Oas3).info.title
+        }'. ${error.message}`
+      )
+    }
```

Comment:
I think you should instead provide an option to allow the bypass the validator. Allowing to proceed with what it believes to be invalid openapi should not be the default. You should have to flip a switch if you want to use an invalid document. I think this should be the case because openapi-to-graphql depends on a proper structure and adherence to the spec to produce correct output.

Reply:
This stood out to me because, "tests weren't working so I skipped them" is often not the correct answer. Perhaps you can provide some context about why your validation was failing?

Reply:
@FallingSnow 

> I think you should instead provide an option to allow the bypass the validator.

#409 allows to bypass the validator. It will be pulled in after this one.

> This stood out to me because, "tests weren't working so I skipped them" is often not the correct answer.

I also had similar concerns. @ardatan told me that some of our test OASs are actually invalid, which I think is very possible. I proposed we file an issue and merge this in in the meantime to move things along.

Reply:
I broke the tests again intentionally.